File: frontend/src/types/task.ts
export type TaskStatus = "pending" | "running" | "completed" | "failed" | "cancelled" | string;
export type TaskLogStatus = "started" | "processing" | "completed" | "error" | string;

export interface Task {
  id: string;
  team_id: string;
  user_id: string;
  type: string;
  input: Record<string, unknown>;
  options: Record<string, unknown>;
  status: TaskStatus;
  progress: number;
  started_at: string | null;
  completed_at: string | null;
  created_at: string;
  team_name: string;
  duration: string | null;
}

export interface TaskListResponse {
  tasks: Task[];
  total: number;
  page: number;
  limit: number;
}

export interface TaskCreateInput {
  type: string;
  input?: Record<string, unknown>;
  options?: Record<string, unknown>;
}

export interface TaskLog {
  id: string;
  task_id: string;
  agent_id: string;
  status: TaskLogStatus;
  message: string;
  progress: number;
  created_at: string;
}

export interface TaskLogListResponse {
  logs: TaskLog[];
  total: number;
}

export interface TaskResult {
  id: string;
  task_id: string;
  agent_id: string;
  result_type: string;
  content: string;
  file_url: string | null;
  metadata: Record<string, unknown>;
  quality_score: number | null;
  version: number;
  created_at: string;
}

export interface TaskResultListResponse {
  results: TaskResult[];
  total: number;
}

File: backend/app/schemas/task.py
from datetime import datetime
from typing import Any, Optional

from pydantic import BaseModel, Field

from app.models.task import TaskStatusEnum


class TaskCreate(BaseModel):
    type: str = Field(..., min_length=1, max_length=50)
    input: dict[str, Any] = Field(default_factory=dict)
    options: dict[str, Any] = Field(default_factory=dict)


class TaskResponse(BaseModel):
    id: str
    team_id: str
    user_id: str
    type: str
    input: dict[str, Any]
    options: dict[str, Any]
    status: str
    progress: int
    started_at: Optional[datetime] = None
    completed_at: Optional[datetime] = None
    created_at: datetime
    team_name: str
    duration: Optional[str] = None


class TaskListResponse(BaseModel):
    tasks: list[TaskResponse]
    total: int
    page: int
    limit: int


class TaskListFilters(BaseModel):
    team_id: Optional[str] = None
    status: Optional[TaskStatusEnum] = None
    type: Optional[str] = None
    created_from: Optional[datetime] = None
    created_to: Optional[datetime] = None
    search: Optional[str] = None
    page: int = Field(default=1, ge=1)
    limit: int = Field(default=20, ge=1, le=100)
